qtfred/src/mission/dialogs/CampaignEditorDialogModel.cpp

```diff
@@ -0,0 +1,801 @@
+	if (campaignType == campaignTypes[0])
+		return 0;
+	for (const auto *mn : missionData.getCheckedData())
+		if (mn->filename == firstMission)
+			return mn->nPlayers;
+	return 0;
+}
+
+void CampaignEditorDialogModel::connectBranches(bool uiUpdate, const campaign *cpgn) {
+	for (auto *mn: missionData.getCheckedData()) {
+		if (cpgn) {
+			const cmission *const cm_it{
+				std::find_if(cpgn->missions, &cpgn->missions[cpgn->num_missions],
+						[&](const cmission &cm){ return mn->filename == cm.name; })};
+			if (cm_it != &cpgn->missions[cpgn->num_missions]) {
+				mn->branchesFromFormula(this, cm_it->formula);
+				mn->branchesFromFormula(this, cm_it->mission_loop_formula, cm_it);
+			}
+		}
+		for (auto& br: mn->branches)
+			if (br.type == CampaignBranchData::NEXT || br.type == CampaignBranchData::NEXT_NOT_FOUND)
+				br.connect(const_cast<const CheckedDataListModel<CampaignMissionData>&>(missionData).getCheckedData());
+	}
+	if (uiUpdate)
+		parent->updateUIMission();
+}
+
+bool CampaignEditorDialogModel::fillTree(sexp_tree& sxt) const {
+	if (!mnData_it) return false;
+	int i = 0;
+	for (const CampaignBranchData& br : mnData_it->branches) {
+		NodeImage img;
+		if (br.type == CampaignBranchData::NEXT_NOT_FOUND)
+			img = NodeImage::ROOT_DIRECTIVE;
+		else if (br.loop)
+			img = NodeImage::ROOT;
+		else
+			img = NodeImage::BLACK_DOT;
+		QTreeWidgetItem *h = sxt.insert(CampaignBranchData::branchTexts.at(br.type) + br.next, img);
+		h->setData(0, Qt::UserRole, i++);
+		sxt.add_sub_tree(sxt.load_sub_tree(br.sexp, true, "do-nothing"), h);
+	}
+	mnData_it->brData_it = nullptr;
+	mnData_it->brData_idx = -1;
+	return true;
+}
+
+void CampaignEditorDialogModel::supplySubModelLoop(QPlainTextEdit &descr) {
+	if (getCurBr())
+		getCurBr()->loopDescr->associateEdit(&descr);
+	else
+		descr.setDocument(nullptr);
+}
+
+bool CampaignEditorDialogModel::saveTo(const QString &file) {
+	bool success = _saveTo(file);
+	QMessageBox::information(parent, file, success ? tr("Successfully saved") : tr("Error saving"));
+	modified = ! success;
+	return success;
+}
+
+void CampaignEditorDialogModel::trackMissionUncheck(const QModelIndex &idx, const QModelIndex &bottomRight, const QVector<int> &roles) {
+	Assert(idx == bottomRight);
+	Assert(missionData.internalData(idx));
+
+	if (roles.contains(Qt::CheckStateRole)) {
+		auto mn = missionData.internalData(idx);
+		bool checked = missionData.data(idx, Qt::CheckStateRole).toBool();
+
+		//must always have first mission, or no missions
+		if (! missionData.getCheckedData().empty()) {
+			if (! checked) {
+				if (mn->filename == firstMission) {
+					QMessageBox::information(parent, tr("First Mission"), tr("You cannot remove the first mission of a campaign,\nunless it is the only one. Choose another to be first."));
+					missionData.setData(idx, Qt::Checked, Qt::CheckStateRole);
+					return;
+				}
+			} else	if (firstMission == "" && missionData.getCheckedData().size() == 1)
```

Comment:
This else-if is buried in other if statements with a different brace style.  Please use their brace style instead (and enclose the one-liner with `{ }`)